template/dogfood/flake.nix

```diff
@@ -0,0 +1,22 @@
+{
+  description = "Description for the project";
+
+  inputs = {
+    nixpkgs_22_11.url = "github:NixOS/nixpkgs/nixos-22.11";
+  };
+
+  outputs = { flake-parts, self, ... }@inputs:
+    flake-parts.lib.mkFlake { inherit inputs; } (
+      flake-parts.lib.mkFlake
+        {
+          inherit inputs;
+          # Workaround for https://github.com/hercules-ci/flake-parts/issues/148
+          self = {
+            outPath = ./.;
+            inherit (self)
+              _type inputs lastModified lastModifiedDate narHash outputs sourceInfo submodules;
+          };
```

Comment:
This breaks normal use of `self`. I don't think importing from `self` is important enough of a problem.
Making assumptions about the user's file layout is too magical. You are free to implement your own conventions using the `specialArgs` solution instead.